ocaml/forkexecd/helper/vfork_helper.c

```diff
@@ -335,14 +335,43 @@ reset_signal_handlers(void)
 static void
 clear_cgroup(void)
 {
-    int fd = open("/sys/fs/cgroup/systemd/cgroup.procs", O_WRONLY|O_CLOEXEC);
-    if (fd >= 0) {
-        char string_pid[32];
-        int ignored __attribute__((unused));
-        sprintf(string_pid, "%d\n", (int) getpid());
-        ignored = write(fd, string_pid, strlen(string_pid));
+    // list of files to try, terminated by an empty string
+    static const char cgroup_files[] =
+        "/sys/fs/cgroup/systemd/cgroup.procs\0"
+        "/sys/fs/cgroup/cgroup.procs\0"
+        "\0";
+
+    char string_pid[32];
+    int last_error = 0;
+    const char *last_error_msg = NULL;
+    const char *last_fn = NULL;
+
+    sprintf(string_pid, "%d\n", (int) getpid());
+
+    for (const char *fn = cgroup_files; *fn != '\0'; fn += strlen(fn) + 1) {
```

Comment:
Why can't we iterate over the array and need all this character-level logic?

Reply:
Not hard and much more compact in memory. And easy to extend passing a string pointer from outside if needed.

Reply:
Yeah, i agree.  `const char *const files[] = { "...", "...", NULL };` and iterating over this will be simpler.

Reply:
Well, it's a pretty established pattern on Windows, but if we don't mind the extra 100 bytes or so, fine with me.

Reply:
Well, if that.  It's +24b of datastructure, but a simplification of the loop logic which could quite easily be more.  Not that this matters.

What matters is that's easier to read/follow for non-expert-C programmers, which is relevant to maintainability in this case.

Reply:
You are forgetting relocation data :smile: